pkg/endpoint/policy.go

```diff
@@ -481,6 +482,19 @@ func (e *Endpoint) Regenerate(regenMetadata *regeneration.ExternalRegenerationMe
 
 		var buildSuccess bool
 		var regenError error
+		defer func() {
+			if !buildSuccess {
```

Comment:
This is currently also triggered when the build was canceled. Is it intended to retry in that case as well? Do we rely on the endpoint being deleted and thus cancelling the retry controller before the API to create the endpoint returns failure?

Reply:
Will add a check for if the build was canceled here.